docs/automobile/vehicle-generic.md

```diff
@@ -0,0 +1,587 @@
+
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+## TractorFrontWheel
+
+Tractor front wheel.
+
+%figure
+
+![TractorFrontWheel](images/generic/TractorFrontWheel.thumbnail.png)
+
+%end
+
+Derived from [Slot](../reference/slot.md).
+
+```
+TractorFrontWheel {
+  SFString name           "tractor front wheel"
+  SFFloat  tireRadius     0.38
+  SFFloat  width          0.19
+  SFInt32  subdivision    24
+  SFBool   wheelSide      FALSE
+  SFNode   physics        Physics { density -1 mass 30 }
+  SFBool   boundingObject TRUE
+}
+```
+
+> **File location**: "[WEBOTS\_HOME/projects/vehicles/protos/generic/TractorFrontWheel.proto]({{ url.github_tree }}/projects/vehicles/protos/generic/TractorFrontWheel.proto)"
+
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+## TractorRearWheel
+
+Tractor rear wheel.
+
+%figure
+
+![TractorRearWheel](images/generic/TractorRearWheel.thumbnail.png)
+
+%end
+
+Derived from [Slot](../reference/slot.md).
+
+```
+TractorRearWheel {
+  SFString name           "tractor rear wheel"
+  SFFloat  tireRadius     0.6
+  SFFloat  width          0.37
+  SFInt32  subdivision    24
+  SFBool   wheelSide      FALSE
+  SFNode   physics        Physics { density -1 mass 60 }
+  SFBool   boundingObject TRUE
+}
+```
+
+> **File location**: "[WEBOTS\_HOME/projects/vehicles/protos/generic/TractorRearWheel.proto]({{ url.github_tree }}/projects/vehicles/protos/generic/TractorRearWheel.proto)"
+
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+## Truck
+
+A simple truck.
+
+%figure
+
+![Truck](images/generic/Truck.thumbnail.png)
+
+%end
+
+Derived from [Robot](../reference/robot.md).
+
+```
+Truck {
+  SFVec3f    translation       0 0 0.5
+  SFRotation rotation          0 0 1 0
+  SFColor    color             0.39 0 0
+  SFNode     trailer           TruckTrailer { }
+  SFFloat    wheelbase         5.25
+  SFFloat    kingPinDistance   1.35
+  SFFloat    mass              8000
+  SFString   name              "vehicle"
+  SFString   controller        "<generic>"
+  MFString   controllerArgs    [ ]
+  SFBool     supervisor        FALSE
+  SFBool     synchronization   TRUE
+  MFNode     sensorsSlotFront  [ ]
+  MFNode     sensorsSlotRear   [ ]
+  MFNode     sensorsSlotTop    [ ]
+  MFNode     sensorsSlotCenter [ ]
+  SFString   window            "automobile"
+}
+```
+
+> **File location**: "[WEBOTS\_HOME/projects/vehicles/protos/generic/Truck.proto]({{ url.github_tree }}/projects/vehicles/protos/generic/Truck.proto)"
+
+> **License**: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+[More information.](https://cyberbotics.com/webots_assets_license)
+
+### Truck Field Summary
```

Comment:
Why is there a field summary only for the Truck and for the TruckTrailer? It doesn't seem very consistent. The Bus doesn't have a summary for example.

Reply:
You are right.
All the `vehicles-*.md` files are automatically generated by the script, but it seems indeed wrong that for example the Bus.color field is not documented.
I will check the documentation script.

Reply:
I'm adding the comments in the PROTO files so that the documentation is generated correctly.